flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/BoundedBlockingSubpartitionReader.java

```diff
@@ -67,6 +73,25 @@
 		this.dataBufferBacklog = numDataBuffers;
 
 		this.nextBuffer = dataReader.nextBuffer();
+		this.availabilityListener = checkNotNull(availabilityListener);
+	}
+
+	BoundedBlockingSubpartitionReader(
```

Comment:
please re-use one of the constructors in the the other

Reply:
I actually considered this issue. But because of `data.createReader(this)` we could not re-use another constructor directly via `this`. Or we define another `initialize` method for extracting the common codes of two constructors?

Reply:
Only retain one constructor for fixing.